p2p/conn/connection.go

```diff
@@ -315,7 +337,18 @@ FOR_LOOP:
 			c.Logger.Debug("Send Ping")
 			wire.WriteByte(packetTypePing, c.bufWriter, &n, &err)
 			c.sendMonitor.Update(int(n))
+			c.Logger.Debug("Starting pong timer", "dur", c.config.PongTimeout)
+			c.pongTimer = time.AfterFunc(c.config.PongTimeout, func() {
+				c.pongTimeoutCh <- true
```

Comment:
this could/should be non-blocking.  probably select on .quit

Reply:
hmm - won't that cause us to miss a lot of these ? if we're sending a peer lots of data, but it never responds to pings, we might never detect that.